### ISSUE TITLE
Double parenthesis breaks parsing

### ISSUE THREAD

**Issue by ying-w** [sqlfluff/sqlfluff#4450]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fails the parse the file when there are multiple parenthesis together

```sql
insert into myproject.tmp.table (
 (
 SELECT 1
)
)
```

This is useful when you have a DDL statement like

```sql
INSERT INTO mytable (
  (
    SELECT
      col1,
      col2
    FROM table1
  )
  UNION ALL
  (
    SELECT
      col1,
      col2
    FROM table2
  )
)
```

### Expected Behaviour

Sqlfluff should get rid of the extra parenthesis

### Observed Behaviour

Parsing violation, expected statement

### How to reproduce

```
% sqlfluff parse tmp2.sql -v
==== sqlfluff ====
sqlfluff:                1.4.5 python:                  3.9.6
implementation:        cpython verbosity:                   1
dialect:              bigquery templater:               jinja
rules:                                all
=== [ path: tmp2.sql ] ===

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'statement'
[L:  1, P:  1]      |            raw:                                              'insert'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            raw:                                              'into'
[L:  1, P: 12]      |            whitespace:                                       ' '
[L:  1, P: 13]      |            raw:                                              'myproject'
[L:  1, P: 22]      |            raw:                                              '.'
[L:  1, P: 23]      |            raw:                                              'tmp'
[L:  1, P: 26]      |            raw:                                              '.'
[L:  1, P: 27]      |            raw:                                              'table'
[L:  1, P: 32]      |            whitespace:                                       ' '
[L:  1, P: 33]      |            raw:                                              '('
[L:  1, P: 34]      |            newline:                                          '\n'
[L:  2, P:  1]      |            whitespace:                                       ' '
[L:  2, P:  2]      |            raw:                                              '('
[L:  2, P:  3]      |            newline:                                          '\n'
[L:  3, P:  1]      |            whitespace:                                       ' '
[L:  3, P:  2]      |            raw:                                              'SELECT'
[L:  3, P:  8]      |            whitespace:                                       ' '
[L:  3, P:  9]      |            numeric_literal:                                  '1'
[L:  3, P: 10]      |            newline:                                          '\n'
[L:  4, P:  1]      |            raw:                                              ')'
[L:  4, P:  2]      |            newline:                                          '\n'
[L:  5, P:  1]      |            raw:                                              ')'
[L:  5, P:  2]      |    newline:                                                  '\n'
[L:  6, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'insert into
                       | myproject.tmp.table (\n (\n SE...'
                       ```

```

file

```
% cat tmp2.sql
insert into myproject.tmp.table (
 (
 SELECT 1
)
)
```

### Dialect

bigquery

### Version

1.4.5

### Configuration
```
[sqlfluff]
dialect = bigquery

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 88
indent_unit = space

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L064]
# Consistent usage of preferred quotes for quoted literals
preferred_quoted_literal_style = single_quotes
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by yoichi** [comment]
I think the parse error will be fixed with #4504.

> Sqlfluff should get rid of the extra parenthesis

Currently, there is no rule to get rid of the extra parenthesis. 
It seems possible to add a rule.

**Comment by yoichi** [comment]
> > Sqlfluff should get rid of the extra parenthesis
> 
> Currently, there is no rule to get rid of the extra parenthesis. It seems possible to add a new rule.

After doing a bit of research,  I think this rule is pretty hard to implement, since double parentheses are required in some places. For example, functional key parts in CREATE INDEX syntax for MySQL:

```
CREATE INDEX idx1 ON t1 ((col1 + col2));
```
https://dev.mysql.com/doc/refman/8.0/en/create-index.html#create-index-functional-key-parts

**Comment by ying-w** [comment]
> After doing a bit of research, I think this rule is pretty hard to implement, since double parentheses are required in some places. For example, functional key parts in CREATE INDEX syntax for MySQL:

makes sense, thanks for looking into it
